File: java/Event.java
package dita.dev.myportal.ui.base.v2;

import android.arch.lifecycle.LifecycleOwner;

public class Event extends SingleLiveEvent<Void> {

    public void observe(LifecycleOwner owner, final EventObserver observer) {
        super.observe(owner, s -> {
            observer.onEvent();
        });
    }

    public interface EventObserver {
        void onEvent();
    }
}

File: kotlin/Event.kt
package dev.dita.maziwapp.ui.base.v2

import android.arch.lifecycle.LifecycleOwner
import android.arch.lifecycle.Observer


class Event : SingleLiveEvent<Void>() {
    open fun observe(owner: LifecycleOwner, observer: EventObserver) {
        super.observe(owner, Observer {
            observer.onEvent()
        })
    }

    open interface EventObserver {
        fun onEvent()
    }
}